src/store.ts

```diff
@@ -934,6 +949,54 @@ const createMlStore = (logging: Logging) => {
           setPostImportDialogState(state: PostImportDialogState) {
             set({ postImportDialogState: state });
           },
+
+          startPredicting(buffer: BufferedData) {
+            const { actions, model, predictionInterval, dataWindow } = get();
+            if (!model || predictionInterval) {
+              return;
+            }
+            const newPredictionInterval = setInterval(() => {
+              const startTime = Date.now() - dataWindow.duration;
+              const input = {
+                model,
+                data: buffer.getSamples(startTime),
+                classificationIds: actions.map((a) => a.ID),
+              };
+              if (input.data.x.length > dataWindow.minSamples) {
+                const result = predict(input, dataWindow);
+                if (result.error) {
+                  logging.error(result.detail);
+                } else {
+                  const { confidences } = result;
+                  const detected = getDetectedAction(
+                    // Get latest actions from store so that changes to
+                    // recognition point are realised.
+                    get().actions,
```

Comment:
This seems to work, but unsure if there's a better way to manage this.

Reply:
I think this access is OK. We do something similar to get the latest project but I kinda prefer how that works by calling API on the store rather than being inside the store. Not pushing to change it now though. I'd like time to tidy the store post release.